covalent/_results_manager/result.py

```diff
@@ -455,24 +454,68 @@ def _write_dispatch_to_python_file(self, directory: str = None) -> None:
         # Accumulate the tasks and workflow in a string
         topo_sorted_graph = self.lattice.transport_graph.get_topologically_sorted_graph()
         functions_added = []
+        serialized_functions = ""
         for level in topo_sorted_graph:
             for nodes in level:
                 function = self.lattice.transport_graph.get_node_value(
                     nodes, value_key="function"
                 ).get_deserialized()
                 if function is not None and function.__name__ not in functions_added:
 
-                    dispatch_function += self.lattice.transport_graph.get_node_value(
+                    serialized_function = self.lattice.transport_graph.get_node_value(
                         nodes, value_key="function_string"
                     )
+                    serialized_functions += serialized_function
                     functions_added.append(function.__name__)
 
         lattice_function_str = convert_to_lattice_function_call(
             self.lattice.workflow_function_string,
             self.lattice.workflow_function.__name__,
             self.inputs,
         )
-        dispatch_function += lattice_function_str
+
+        dispatch_function = dispatch_header + serialized_functions + lattice_function_str
+        dispatch_function = self._remove_import_dupes(dispatch_function)
 
         with open(os.path.join(result_folder_path, "dispatch_source.py"), "w") as f:
             f.write(dispatch_function)
+
+    @staticmethod
+    def _remove_import_dupes(function_string: str) -> str:
+        """
+        Inspect a snippet of code and remove any duplicate imports
+
+        Args:
+            function_string: The input code snippet as a string.
+            imports: A list of already-collected imports.
+
+        Returns:
+            The code snippet with any duplicate imports removed.
+        """
+        imports = set()
+        function_lines = function_string.split("\n")
+        dupes = []
+        for i in range(len(function_lines)):
+            line = function_lines[i].strip().lstrip("#").lstrip()
+            row = line.split()
+            # imports in the serialized function have a limitied number of forms.
+            # None are multiple imports in one line.
+            if (
+                (len(row) == 2 and row[0] == "import")  # import x
+                or (len(row) == 4 and row[0] == "import" and row[2] == "as")  # import x as y
+                or (len(row) == 4 and row[0] == "from" and row[2] == "import")  # from x import y
+                or (
+                    len(row) == 6 and row[0] == "from" and row[2] == "import" and row[4] == "as"
+                )  # from x import y as z
+            ):
+                if line in imports:
+                    dupes.append(i)
+                else:
+                    imports.add(line)
+        if len(dupes) > 0:
```

Comment:
Also this as `if dupes`.

Reply:
I prefer `len(dupes)` because it lets the reader quickly know that dupes is a container which should not be empty. As opposed to a bool or some other object that could return True/False when asked `if dupes`

Reply:
Ah, I see your point.